.github/workflows/test.yml

```diff
@@ -38,17 +42,13 @@ jobs:
     runs-on: ubuntu-latest
     steps:
     - uses: actions/checkout@v4
-      with:
-        fetch-depth: 0
```

Comment:
We don't need to fetch the entire git history in any of these jobs.

Reply:
You do because you need the git-rev-parse to work for the hatch-vcs version.

Reply:
You could argue that we don't need it for all jobs, but then we end up with weird glass-0.1something versions in some tests, which I would find confusing. (As a matter of fact, I only recently fixed the fetch-depth because I couldn't understand why the action didn't test the latest version.)

Reply:
> We do because we need the git-rev-parse to work for the hatch-vcs version.

I meant that we don't need hatch-vcs to do the versioning work for these job, but I see, it can be confusing when reading the workflow logs. I'll revert it!